src/main/java/mops/businesslogic/FileInfoServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package mops.businesslogic;
+
+import lombok.AllArgsConstructor;
+import mops.exception.MopsException;
+import mops.persistence.FileInfoRepository;
+import mops.persistence.file.FileInfo;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@AllArgsConstructor
+public class FileInfoServiceImpl implements FileInfoService {
+
+    /**
+     * reference to fileInfoRepository.
+     */
+    private final FileInfoRepository fileInfoRepo;
+
+    /**
+     * @param dirId directory id
+     * @return a list of files in that directory
+     */
+    @Override
+    public List<FileInfo> fetchAllFilesInDirectory(long dirId) throws MopsException {
+        return fileInfoRepo.getAllFileInfoByDirectory(dirId);
+    }
+
+    /**
+     * @param fileId file id
+     * @return a FileInfo object
+     */
+    @SuppressWarnings("PMD")
+    @Override
+    public FileInfo fetchFileInfo(long fileId) throws MopsException {
+        return fileInfoRepo.findById(fileId).orElseThrow(() -> new MopsException("..."));
+    }
+
+    /**
+     * @param fileInfo Metadata of a file
+     * @return ID the FileInfo was saved under
+     */
+    @SuppressWarnings("PMD")
+    @Override
+    public FileInfo saveFileInfo(FileInfo fileInfo) throws MopsException {
+        try {
+            return fileInfoRepo.save(fileInfo);
+        } catch (Exception e) {
+            //TODO: better exception
+            throw new MopsException("...", e);
```

Comment:
indeed

Reply:
@Segelzwerg I implemented the DataAccessException instead of Exception - there are no references in documentation. After googling I found DataAccessException with 4 sub-classes, which are not intuitive to use in this context (for me)

Reply:
you should implement a Custom Exception which inherits from `MopsException` like: [WriteAccessException](https://github.com/hhu-propra2/abschlussprojekt-mopse/blob/master/src/main/java/mops/businesslogic/exception/WriteAccessPermissionException.java)